src/cmd/cmd_c.ml

```diff
@@ -33,6 +33,59 @@ let find location file : Fpath.t Result.t =
   in
   loop l
 
+let eacsl_instrument eacsl debug ~includes (files : Fpath.t list) :
+  Fpath.t list Result.t =
+  if eacsl then
+    let flags1 =
+      let includes =
+        String.concat " "
+          (List.map (fun libpath -> "-I" ^ Fpath.to_string libpath) includes)
+      in
+      let framac_verbosity_level = if debug then "2" else "0" in
+      Cmd.(
+        of_list
+          [ "-e-acsl"
+          ; "-no-frama-c-stdlib"
+          ; "-kernel-warn-key"
+          ; "CERT:MSC:38=inactive"
+          ; "-verbose"
+          ; framac_verbosity_level
+          ; String.concat "" [ "-cpp-extra-args=\""; includes; "\"" ]
+          ] )
+    in
+    let flags2 = Cmd.(of_list [ "-then-last"; "-print"; "-ocode" ]) in
+
+    let* framac_bin = OS.Cmd.resolve @@ Cmd.v "frama-c" in
+
+    let outs =
+      List.map
+        (fun file ->
+          let ext = Fpath.get_ext ~multi:true file in
+          let file_without_ext = Fpath.rem_ext ~multi:true file in
+          let file_without_ext =
+            Fpath.to_string file_without_ext ^ "_instrumented"
+          in
+          let file = file_without_ext ^ ext in
```

Comment:
I don't think this is building with the latest changes adding `prelude`. What do you get when you do `dune build @all` ?

You should avoid the `Fpath.to_string` and `^` stuff, for instance, see how it is done in the `write_file` function from `ast/binary_encoder.ml`.

Reply:
Seems I've not rebased to the lastest commits